lib/dbl.php

```diff
@@ -129,28 +138,25 @@ function sanitized_dsn() {
 
     /** @return ?\mysqli */
     function connect() {
-        assert(($this->name ?? "") !== "");
-        if ($this->socket) {
-            $dblink = new mysqli($this->host, $this->user, $this->password, "", $this->port, $this->socket);
-        } else if ($this->port !== null) {
-            $dblink = new mysqli($this->host, $this->user, $this->password, "", $this->port);
-        } else {
-            $dblink = new mysqli($this->host, $this->user, $this->password);
-        }
+	assert(($this->name ?? "") !== "");
+
+        $dblink = new mysqli();
+        $dblink->ssl_set($this->sslKey, $this->sslCert, $this->sslCaCert, null, null);
+        $dblink->options(MYSQLI_OPT_SSL_VERIFY_SERVER_CERT, $this->sslVerifyServerCert ?? false);
+        $dblink->real_connect($this->host, $this->user, $this->password, $this->name, $this->port, $this->socket);
```

Comment:
Are you sure you don't need MYSQLI_CLIENT_SSL/MYSQLI_CLIENT_SSL_DONT_VERIFY_SERVER_CERT here?

Reply:
as $flags arguments?

Reply:
I'm still evaluating, but I'll probably do it similarly to what has been done here:
https://github.com/codeigniter4/CodeIgniter4/blob/ba75eed0751dd48cb84be6116438632f4b3c9a93/system/Database/MySQLi/Connection.php#L142

Reply:
I did something similar to what has been done above - my tests showed that it works as intended when the server requires X.509 authentication (mTLS), but if only SSL is required it seems as if validation does not always take place. I suppose, however, that that is caused by mysqli itself.